packages/components/src/Button/Button.tsx

```diff
@@ -56,11 +56,11 @@ interface ButtonLinkProps extends ButtonFoundationProps {
 }
 
 interface BaseActionProps extends ButtonFoundationProps {
-  readonly variation?: "work" | "learning" | "subtle";
+  readonly variation?: "work" | "learning" | "subtle" | "destructive";
```

Comment:
Out of curiousity, why do we need to change Button? We still have `DestructiveActionProps` below

Reply:
The destructive variation that we created will be assigned to the variation of the primary action prop of Modal. Which is a weird combo of Button props, that can take either BaseActionProps or Destructive action props but not both.  

![image](https://user-images.githubusercontent.com/84454304/133630943-76a5658a-7f6e-43b8-aa7b-cdbde6630eb6.png)